Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java

```diff
@@ -414,18 +408,43 @@ public void finish() {
     @Override
     protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
         super.onActivityResult(requestCode, resultCode, data);
-        getDropInClient().handleThreeDSecureActivityResult(this, resultCode, data, new DropInResultCallback() {
-            @Override
-            public void onResult(@Nullable DropInResult dropInResult, @Nullable Exception error) {
-                if (dropInResult != null) {
-                    finishWithDropInResult(dropInResult);
-                } else if (error instanceof UserCanceledException) {
-                    dropInViewModel.setUserCanceledError(error);
-                } else {
-                    onError(error);
-                }
-            }
-        });
+        handleActivityResult(requestCode, resultCode, data);
```

Comment:
I think we can even take it a step further and call `getDropInClient()#handleActivityResult` here since we have access to the request codes from within `DropInClient`. Then `DropInActivity` only has to call one method and forward the result to the merchant.

Reply:
Ah got it ok moved this whole method to DropInClient 👍